flask/app/families.py

```diff
@@ -23,3 +25,96 @@ def families_list():
     ]
 
     return jsonify(families)
+
+@app.route("/api/families/<int:id>", methods=["GET"])
+@login_required
+def families_by_id(id: int):
+    family = models.Family.query.filter_by(family_id = id).options(
+        joinedload(models.Family.participants)\
+            .joinedload(models.Participant.tissue_samples)\
+                .joinedload(models.TissueSample.datasets)
+    ).first_or_404()
+
+
+    families = [
+        { 
+            **asdict(family),
+            "participants": [
+            {
+                **asdict(participants),
+                "tissue_samples": [
+                        {
+                            **asdict(tissue_samples),
+                            "datasets": [
+                                {
+                                    **asdict(dataset)
+                                }
+                            
+                                for dataset in tissue_samples.datasets
+                            ]
+                        }
+                        for tissue_samples in participants.tissue_samples
+                    ]
+                }
+            for participants in family.participants
+            ]
+        }
+    ]
+    return jsonify(families)
+
+
+@app.route("/api/families/<int:id>", methods = ["DELETE"])
+@login_required
+@check_admin
+def delete_families(id: int):
+    family = models.Family.query.filter_by(family_id = id).options(
+        joinedload(models.Family.participants)
+    )
+
+    fam_entity = family.first_or_404()
+
+    if len(fam_entity.participants) == 0:
+        try:
+            family.delete()
+            db.session.commit()
+            return 'Deletion successful', 204
+        except:
+            db.session.rollback()
+            return 'Deletion of entity failed!', 422
+    else:
+        return 'Family has participants, cannot delete!', 422
+
+
+@app.route("/api/families/<int:id>", methods = ["PATCH"])
+@login_required
+def edit_families(id: int):
+
+    if not request.json:
+        return "Request body must be JSON", 415
+    
+    try:
+        fam_codename = request.json['family_codename']
+    except KeyError:
+        return "No family codename provided", 400
+
+    family = models.Family.query.get_or_404(id)
+
+    if family.family_codename == fam_codename:
+        return 'Family codename is identical', 403
+
```

Comment:
This isn't needed. HTTP 403 is also related to authentication problems only, not this kind of situation.